fast64_internal/f3d/f3d_material.py

```diff
@@ -2070,13 +2081,29 @@ def get_tex_gen_size(tex_size: list[int | float]):
     return (tex_size[0] - 1) / 1024, (tex_size[1] - 1) / 1024
 
 
+def get_textlut_mode(f3d_mat: "F3DMaterialProperty", none_if_not_set: bool = False):
+    use_dict = all_combiner_uses(f3d_mat)
+    textures = [f3d_mat.tex0] if use_dict["Texture 0"] and f3d_mat.tex0.tex_set else []
+    textures += [f3d_mat.tex1] if use_dict["Texture 1"] and f3d_mat.tex1.tex_set else []
+    if textures:
```

Comment:
this is a confusing way to pass None if the two ci_format attributes are different between the texture properties but only on the UI. I think you should just call out each case so it is more clear what is going on here. Either you inherit the property when it is coming from a tex, it comes from the property, or you return None. If you spell out each case directly in the if statements it is more clear what is going on and why each thing is returned.

```suggestion
ci_formats = [tex.ci_format if tex.tex_format.startswith("CI") for tex in textures]
if not ci_formats and textures:
    return "G_TT_NONE"
if len(ci_formats) == 1 or ci_formats[0] == ci_formats[1]:
    return "G_TT_" + ci_formats[0]
else:
     # I am changing your var `none_if_not_set` name
     return None if inherit_from_tex else f3d_mat.rdp_settings.g_mdsft_texlut
```

Reply:
@jesusyoshi54 your code didn´t have the same functionality, so here is what I did, I renamed ci_formats to tlut_modes and kept the NONE for non ci's, then if there is atleast one mode i compare the first element to the last (I think it keeps the logic explicit without adding another len check?), this shortned the code a lot and still made things more explicit hopefully, if you still dislike it I will change it

Reply:
yeah coding in the git UI is sort of cringe